src/irmin/tree.ml

```diff
@@ -774,32 +774,54 @@ module Make (P : Private.S) = struct
         type acc.
         force:acc force ->
         uniq:uniq ->
-        pre:acc node_fn ->
-        post:acc node_fn ->
+        ?pre:acc node_fn ->
+        ?post:acc node_fn ->
         path:Path.t ->
         ?depth:depth ->
         node:(key -> _ -> acc -> acc Lwt.t) ->
         contents:(key -> contents -> acc -> acc Lwt.t) ->
         t ->
         acc ->
         acc Lwt.t =
-     fun ~force ~uniq ~pre ~post ~path ?depth ~node ~contents t acc ->
+     fun ~force ~uniq ?pre ?post ~path ?depth ~node ~contents t acc ->
       let marks =
         match uniq with
         | `False -> dummy_marks
         | `True -> empty_marks ()
         | `Marks n -> n
       in
+      let to_list path bindings acc = function
+        | None -> Lwt.return acc
+        | Some f ->
+            let bindings = List.of_seq bindings in
+            let s = List.rev_map fst bindings in
```

Comment:
```suggestion
            let s = Seq.fold (fun acc (s, _) -> s :: acc) [] bindings in
```

(I think this is equivalent & avoids a traversal.)

Reply:
indeed, thanks!